### ISSUE TITLE
Object reference not set to an instance of an object.

### ISSUE THREAD

**Issue by DarrylO0** [Kyrodan/KeeAnywhere#349]
I am using KeePass v2.51.1, KeeAnywhere v2.0.3 and KeePassHttp v1.8.42 on Windows 10.

My KeePass database is on OneDrive.

I receive an error on File -> Open -> Open from Cloud Drive

Error getting Root node.
Exception:

Object reference not set to an instance of an object.

Thanks for the assistance.

**Comment by datalabMD** [comment]
Hello, I received the same message. The problem appeared recently. Everything worked fine a week ago.

**Comment by KeeGitUser** [comment]
Same here.  New issue in just the last week.  KeePass v2.51.1  KeeAnywhere v2.0.3
Google Drive works.  Error occurs only when using MS Cloud Drive.

**Comment by lsaudon** [comment]
Same error

**Comment by YanisDev-ui** [comment]
Hello, I have the same error not corrected currently

**Comment by KeeGitUser** [comment]
My issue has finally resolved itself.  Now using Keepass 2.52.  Had to remove OneDrive from the KeeAnywhere account list on each computer & then add the account back.  This didn't work initially so something was updated that resolved the problem.  I believe it to be a OneDrive update & not KeePass or KeeAnywhere.
So...  Delete OneDrive from KeeAnywhere account list.  Run all available Windows updates.  Add OneDrive back.  Let us know if that works.

**Comment by YanisDev-ui** [comment]
> My issue has finally resolved itself. Now using Keepass 2.52. Had to remove OneDrive from the KeeAnywhere account list on each computer & then add the account back. This didn't work initially so something was updated that resolved the problem. I believe it to be a OneDrive update & not KeePass or KeeAnywhere. So... Delete OneDrive from KeeAnywhere account list. Run all available Windows updates. Add OneDrive back. Let us know if that works.

Definitly does not work on my side with the same keepas configuration & steps that you recommended performed.

**Comment by Lukkasss** [comment]
> My issue has finally resolved itself. Now using Keepass 2.52. Had to remove OneDrive from the KeeAnywhere account list on each computer & then add the account back. This didn't work initially so something was updated that resolved the problem. I believe it to be a OneDrive update & not KeePass or KeeAnywhere. So... Delete OneDrive from KeeAnywhere account list. Run all available Windows updates. Add OneDrive back. Let us know if that works.

Worked for me!

@YanisDev-ui did you tried this way:

1. Remove OneDrive account
2. Add the account again
3. restart keepass
4. after login, it will tell you that the cached version was updated and it will push to the remote version.

**Comment by bnosie** [comment]
I continue to get this issue, and sometimes removing the account works for a few weeks, and then pops up again.  Very frustrating, given there haven't been any updates in almost 2 years, and this error has been going on for longer than that.

**Comment by KeeGitUser** [comment]
You are correct.  Things are fine for awhile & then go bad seemingly at random.  Always the OneDrive account.  Deleting & re-adding the account sometimes works, sometimes doesn't.  Wait a week & it works again.  Frustrating.  The last time I re-added the account (early this week) I noticed some new behavior I hadn't seen before.  Not sure I have this exactly correct but it went something like this...  The usual browser message came up saying key had been received, you can close this window.  I did that but then a second pop-up appeared (from OneDrive?) asking for permission.  I've never seen this second pop-up appear before.  Anyone see similar behavior?

**Comment by TngRx** [comment]
> > My issue has finally resolved itself. Now using Keepass 2.52. Had to remove OneDrive from the KeeAnywhere account list on each computer & then add the account back. This didn't work initially so something was updated that resolved the problem. I believe it to be a OneDrive update & not KeePass or KeeAnywhere. So... Delete OneDrive from KeeAnywhere account list. Run all available Windows updates. Add OneDrive back. Let us know if that works.
> 
> Worked for me!
> 
> @YanisDev-ui have you tried this way:
> 
>     1. Remove OneDrive account
> 
>     2. Add the account again
> 
>     3. restart keepass
> 
>     4. after login, it will tell you that the cached version was updated and it will push to the remote version.

This is the way ! Thank you

**Comment by anvgfr** [comment]
Same issue continue to happen randomly, now i have remove the account from KeeAnywhere, readded and got an error on the browser : 
![image](https://github.com/Kyrodan/KeeAnywhere/assets/95424421/97b12acc-c902-4a4a-8544-f196cac74023)

After that i see that the account is added into keeanywhere, do a check and add to reauthorize again without success (check fail each time)

**Comment by KeeGitUser** [comment]
This continues to be an issue for me as well.  I have to delete the OneDrive account in question.  Add the account back in (authorization will fail but the account will be added).  Do a "Check" on the account which will fail but then the offered reauthorization will be successful.  Finely, do NOT use File / Open Recent (it will fail).  Go the long route of File / Open / Open from Cloud Drive...

**Comment by Adambean** [comment]
This problem is very unresolved, even with the workarounds tried above.

Currently if you want to use OneDrive with KeePass your best option is to use a native OneDrive client for your host platform then open your file via that in KeePass as if it were local. (E.g. on Windows that would be via "%UserProfile%\OneDrive\..." typically.)

**Comment by KeeGitUser** [comment]
Agreed, it's only getting worse.  I still can't figure why it works on some of my computers & not others.  Thanks for your workaround though.  Hadn't thought of that.  Great tip.

**Comment by bnosie** [comment]
I continue to get this on every computer I have it installed on.  The latest version is from May 2021, so it appears that this plugin has been abandoned.

**Comment by dwmuller** [comment]
This just started happening to me again after several months of bliss. Twice in a week.
